NEWS.md

```diff
@@ -220,6 +220,9 @@ This section lists changes that do not have deprecation warnings.
     `airybiprimex`, `airyaix`, `airybix`, `airyaiprimex`, `airybiprimex`)
     ([#18050]).
 
+  * When a macro is called in the module in which that macro is defined, global variables
+    in the macro are now correctly resolved in the macro definition environment ([#15850]).
```

Comment:
since this was so breaking, we should mention what the symptoms and fix look like - usually undefined variable errors that didn't occur in 0.5, and a few more calls to `esc`, respectively